Feature/security improvements

st3v3nhunt opened the pull request:
Add security headers via helmet npm module. This results in the score given by https://securityheaders.io from an [F](https://securityheaders.io/?q=https%3A%2F%2Fbeta.nhs.uk%2Fredirects%2Fgps&hide=on&followRedirects=on) to an [A](https://securityheaders.io/?q=https%3A%2F%2Fgp-redirect-pr-48.dev.beta.nhschoices.net%2F&hide=on&followRedirects=on).

I've not added the `Referrer-Policy` header due to possible scenarios requiring information when switching between https://beta.nhs.uk and http://nhs.uk.

st3v3nhunt committed:
:up: Update npm dependencies

st3v3nhunt committed:
:tophat: Add helmet to fix header security issues

adamgraves-choices commented on app.js:
```diff
@@ -14,6 +15,21 @@ promBundle.promClient.collectDefaultMetrics();
+app.use(helmet.hidePoweredBy());
+app.use(helmet.ieNoOpen());
+app.use(helmet.noSniff());
+app.use(helmet.hsts());
```
I'm assuming this is adding the HSTS header?

st3v3nhunt replied:
It is. As a bunch of other apps are already serving it.

st3v3nhunt replied:
As discussed, this module will continue to be run but will be configured to not include subdomains.

st3v3nhunt committed:
:gem: Exclude subdomains

This is important as the top level domain isn't own by NHSUK and
therefore there might be some stuff happening on subdomains we have no
idea or control over.
